addons/mail/tests/common.py

```diff
@@ -1690,9 +1690,12 @@ class MailCommon(MailCase):
     """ Almost-void class definition setting the savepoint case + mock of mail.
     Used mainly for class inheritance in other applications and test modules. """
 
+    @data_depends(
```

Comment:
Did not understand everything, but why are we using a word telling "it depends from something" while it is actually calling those methods in the method ?

Also I don't like removing super, it makes code really hard to follow.

Can't you fix account and let mail addons alone :D ?

Reply:
`data_depends` defines the dependencies used within `setUpCommonData`, so any attribute defined in here that you override will warrant a new branch.

Unfortunately if we want to make it work with accounting overriding it in mail is an obligation too (we don't want to change call order).
We could split parts in `setUpCommonData` and in `setUpClass` but I think it would make it even more confusing.

The issue here is probably that the feature is not clear enough, we definitely need to document it more :)

FWIW this is quite a performance improvement if used correctly:
```
accounting to auth_totp
without common data:
odoo.service.server: 1888 post-tests in 2840.07s, 2107461 queries
with common data:
odoo.service.server: 1888 post-tests in 2080.38s, 1470447 queries
```